recipe/build-llama-cpp.sh

```diff
@@ -2,7 +2,7 @@
 set -ex
 
 if [[ ${gpu_variant:0:5} = "cuda-" ]]; then
-    CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CUDA_ARCHITECTURES=all"
+    CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_CUDA_ARCHITECTURES=all-major"
```

Comment:
@danpetry Yeah, I messed with explicit arch lists for cuda, but further reading and looking at other packages it makes more sense to run with all-major as opposed to all since that covers the "widest compatibility" part without doing micro-versions and others. I'm not quite sure about PTX, I'll have to research what the tradeoffs are. 

One of the things I need to do once this merges is performance benchmarks for all platform/permutations so we can better understand the tradeoffs of things like this.

Reply:
ok, sure. PTX gives the optimizations of minor versions when you don't have a binary compiled for that minor version. It's an intermediate representation which can be compiled to machine code at runtime.

Reply:
@danpetry correct me if I am wrong, but the only way to force PTX is to explicitly list the architectures like pytorch does, or drop back to `all` right?

Reply:
Looking at it a bit closer, I think you can't use `CMAKE_CUDA_ARCHITECTURES` to get PTX compiled anyway. I was just giving some background info on the topic really in case that's useful though.

> One of the things I need to do once this merges is performance benchmarks for all platform/permutations so we can better understand the tradeoffs of things like this.

That seems like a good approach to me :)